### PR TITLE
update components table

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

File: docs/components.md [hunk 1]
```diff
@@ -4,68 +4,117 @@
...
-|-----------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------|------------|
-| [discovery](../internal/receiver/discoveryreceiver) | [logstransform](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/logstransformprocessor) | [pulsar](../internal/exporter/pulsarexporter) |            |
-|                                                     | [span_metrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/spanmetricsprocessor)    |                                               |            |
-|                                                     | [timestamp](../pkg/processor/timestamp)                                                                                       |                                               |            |
+> Each component has a link to its configuration documentation.
+
+<div style="display: grid;grid-template-columns: auto auto auto auto;">
+
+<div>
```
Comment (dmitryax) [inline docs/components.md:11]: Why are these empty div blocks needed?

**Review by dmitryax — APPROVED** [review]
LGTM

File: docs/components.md (hunk 1 above)
Comment (atoulme) [reply docs/components.md:11]: Just playing - it looks like with these in some markdown envs you can display all the tables side by side. Github doesn't seem to agree though.

**Comment by atoulme** [comment]
I didn't expect a quick approval :) thank you!
Here is what this does:
It removes the big table and breaks it into a table per type of component. Much easier to find things and manage.
It updates all components to the proper stability level. We had a few misses and a few wrong stability levels.
It uses the proper stability levels per OpenTelemetry standard, linking to them.

**Pull request merged**